src/rgw/rgw_iam_policy.cc

```diff
@@ -224,7 +224,7 @@ boost::optional<ARN> ARN::parse(const string& s, bool wildcards) {
 			     std::regex_constants::ECMAScript |
 			     std::regex_constants::optimize);
   static const regex rx_no_wild(
-    "arn:([^:*]*):([^:*]*):([^:*]*):([^:*]*):([^:*]*)",
+    "arn:([^:*]*):([^:*]*):([^:*]*):([^:*]*):(.*)",
```

Comment:
I changed this to allow port numbers in the IDP url given in the trust policy, although AWS docs state that port numbers shouldnt be allowed. But AWS doc also states that the IDP Url should be the same as the 'iss' field in the token, which for my keycloak server is: https://localhost:8080/auth/realms/demo:app_id. Please advise on which approach is correct.

Reply:
@pritha-srivastava I'm not sure;  it sounds like allowing a non-standard port is appropriate, to me

Reply:
Does Keycloak always run on a nonstandard port like that or is it just an artifact of how it's installed? (Is it expecting to be forwarded by another server/)

Either way, it seems reasonable to allow a port.

Reply:
Keycloak by default comes up on 8080 for an http connection https://www.keycloak.org/docs/2.5/server_installation/topics/network/ports.html. The port number can be changed while starting Keycloak.